osu.Game/Skinning/SkinManager.cs

```diff
@@ -175,6 +175,13 @@ private void populateMetadata(SkinInfo item, Skin instance)
                 item.Name = item.Name.Replace(".osk", "", StringComparison.OrdinalIgnoreCase);
                 item.Creator ??= unknown_creator_string;
             }
+
+            // generally when importing from a folder, the ".osk" extension will not be present.
+            // if we ever need a more reliable method of determining this, the type of `ArchiveReader` can be checked.
+            bool isArchiveImport = archiveName?.Contains(".osk", StringComparison.OrdinalIgnoreCase) == true;
+
+            if (archiveName != null && !isArchiveImport && archiveName != item.Name)
```

Comment:
I intentionally opted not to do this. Skins downloaded should have correct metadata. If they don't, that's a separate issue which we can deal with separately.

Reply:
But that means one could hit the same issue as in the thread, but with `.osk`s rather than direct stable imports. It is technically a different issue and potentially debatable, but thought it may be worth mentioning it here.

Reply:
I don't really have a strong opinion on this but I'll just link https://github.com/ppy/osu/issues/10115, as it is a prior issue report that explicitly requests for skin metadata to be used for de-duplicating purposes (i.e. in case of skin updates).

Reply:
Keep in mind the main request there was for overwriting existing, which we still don't really handle "correctly". I don't think this PR changes their use case much.

Reply:
Yeah, and my suggestion is only to allow displaying a "more readable" name next to the metadata name, so that the user is aware where the skin is on the dropdown when importing from online. But I've been testing with one popular skin only, so I've grep'd all names and authors of the skins I had, and it doesn't seem *that* bad to use metadata name on osk import:
```
[I] salman@Salmans-MBP ~/Desktop (master)> grep -E -a "Name|Author" ~/.osu/Skins/*/skin.ini
/Users/salman/.osu/Skins/-+ Seoul v9 -17-03/skin.ini:Name: -+ Seoul v9 -17-03
/Users/salman/.osu/Skins/-+ Seoul v9 -17-03/skin.ini:Author: Seouless
/Users/salman/.osu/Skins/-+ Seoul v9 Personal Skin Nomod/skin.ini:Name: -+ Seoul v9 Personal Skin Nomod
/Users/salman/.osu/Skins/-+ Seoul v9 Personal Skin Nomod/skin.ini:Author: Seouless + Various Artists
/Users/salman/.osu/Skins/Cookiezi 29 2018-03-18/skin.ini:Name: -+ Seoul v9 Personal Skin // Mika Pikazo
/Users/salman/.osu/Skins/Cookiezi 29 2018-03-18/skin.ini:Author: Seouless
/Users/salman/.osu/Skins/FlyingTuna 2018-02-15 Selyu v2.2/skin.ini:Name: - Selyu v2.2
/Users/salman/.osu/Skins/FlyingTuna 2018-02-15 Selyu v2.2/skin.ini:Author: Selyu (I stole all these elements + Cursors)
/Users/salman/.osu/Skins/KeigoClearHandCursor-~2012/skin.ini:Name: ClearSkin (KeigoClear)
/Users/salman/.osu/Skins/KeigoClearHandCursor-~2012/skin.ini:Author: Xiaounlimited
/Users/salman/.osu/Skins/New Retro Wave/skin.ini:Name: - - New Retro Wave Skin -
/Users/salman/.osu/Skins/New Retro Wave/skin.ini:Author: Sv.Vadimka, Mira
/Users/salman/.osu/Skins/Rafis 2018-03-26 HDDT/skin.ini:Name: Prawilosc vJP without followpoints
/Users/salman/.osu/Skins/Rafis 2018-03-26 HDDT/skin.ini:Author: DDK RPK
/Users/salman/.osu/Skins/rrtyui 2017-08-10/skin.ini:Name: rrtyuiSkin
/Users/salman/.osu/Skins/rrtyui 2017-08-10/skin.ini:Author: rrtyui
/Users/salman/.osu/Skins/Seoul++ 16-9 X Ryuk/skin.ini:Name: Seoul++ 16-9 X Ryuk
/Users/salman/.osu/Skins/Seoul++ 16-9 X Ryuk/skin.ini:Author: Seouless + Various Artists
/Users/salman/.osu/Skins/Seouless-2017-08-10/skin.ini:Name: - ShigeSeoul(Seouless v4)
/Users/salman/.osu/Skins/Seouless-2017-08-10/skin.ini:Author: Various Artists
```

so I'll leave that as-is until someone complains about it.